provider/cluster_waiter_resource.go

```diff
@@ -0,0 +1,164 @@
+	nonPositiveTimeoutFormat  = "Can't poll state of cluster with identifier '%s', the timeout that was set is not a positive number"
+	pollingIntervalInMinutes  = 1
+)
+
+func (t *ClusterWaiterResourceType) GetSchema(ctx context.Context) (result tfsdk.Schema,
+	diags diag.Diagnostics) {
+	result = tfsdk.Schema{
+		Description: "Wait Cluster Resource To be Ready",
+		Attributes: map[string]tfsdk.Attribute{
+			"cluster": {
+				Description: "Identifier of the cluster.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"timeout": {
+				Description: "An optional timeout till the cluster is ready. The timeout value should be in minutes." +
+					" the default value is 60 minutes",
+				Type:     types.Int64Type,
+				Optional: true,
+			},
+			"ready": {
+				Description: "Whether the cluster is ready",
+				Type:        types.BoolType,
+				Computed:    true,
+			},
+		},
+	}
+	return
+}
+
+func (t *ClusterWaiterResourceType) NewResource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.Resource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation: use it directly when needed.
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	collection := parent.connection.ClustersMgmt().V1().Clusters()
+
+	// Create the resource:
+	result = &ClusterWaiterResource{
+		logger:     parent.logger,
+		collection: collection,
+	}
+	return
+}
+
+func (r *ClusterWaiterResource) Create(ctx context.Context,
+	request tfsdk.CreateResourceRequest, response *tfsdk.CreateResourceResponse) {
+	// Get the plan:
+	state := &ClusterWaiterState{}
+	diags := request.Plan.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	state.Ready = types.Bool{
+		Value: false,
+	}
+
+	timeout := defaultTimeoutInMinutes
+	if !state.Timeout.Unknown && !state.Timeout.Null {
+		if state.Timeout.Value <= 0 {
+			response.Diagnostics.AddWarning(nonPositiveTimeoutSummary, fmt.Sprintf(nonPositiveTimeoutFormat, state.Cluster.Value))
+		} else {
+			timeout = state.Timeout.Value
+		}
+	}
+
+	// Wait till the cluster is ready:
+	object, err := r.isClusterReady(state.Cluster.Value, ctx, timeout)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't poll cluster state",
+			fmt.Sprintf(
+				"Can't poll state of cluster with identifier '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+	isClusterReady := false
+	if object.State() == cmv1.ClusterStateReady {
+		isClusterReady = true
+	}
+
+	state.Ready = types.Bool{
+		Value: isClusterReady,
+	}
+
+	// Save the state:
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterWaiterResource) Read(ctx context.Context, request tfsdk.ReadResourceRequest,
+	response *tfsdk.ReadResourceResponse) {
+	// Do Nothing
+}
+
+func (r *ClusterWaiterResource) Update(ctx context.Context, request tfsdk.UpdateResourceRequest,
```

Comment:
Shouldn't update also have the same logic as Create? 
What happen if the cluster is not ready when update is performed?

Reply:
The waiting is only in destroy..

Reply:
I thought you wanted me to add an update support for the *new* resource, but I think you talked about cluster update - so in cluster update, it will not affect that resource at all - while updating a cluster, the resource cluster_waiter will be successfully created...